File: staging_vespalib/src/vespa/vespalib/metrics/gauge.cpp
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
#include "gauge.h"
#include "metrics_manager.h"

namespace vespalib {
namespace metrics {

void
Gauge::sample(double value, Point point) const
{
    if (_manager) {
        _manager->sample(Measurement(std::make_pair(_id, point), value));
    }
}

} // namespace vespalib::metrics
} // namespace vespalib

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/container/metrics/Gauge.java
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.vespa.hosted.node.admin.container.metrics;

/**
 * @author freva
 */
public class Gauge implements MetricValue {
    private final Object lock = new Object();

    private double value;

    public void sample(double x) {
        synchronized (lock) {
            this.value = x;
        }
    }

    @Override
    public Number getValue() {
        synchronized (lock) {
            return value;
        }
    }
}
